File: SDKs/exavault.com/2.0/typescript/src/sdk/models/operations/getresourceinfo.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetResourceInfoQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=include" })
  include?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=resource" })
  resource: string;
}


export class GetResourceInfoHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=ev-access-token" })
  evAccessToken: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=ev-api-key" })
  evApiKey: string;
}


export class GetResourceInfoRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetResourceInfoQueryParams;

  @SpeakeasyMetadata()
  headers: GetResourceInfoHeaders;
}


export class GetResourceInfoResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  resourceResponse?: shared.ResourceResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/exavault.com/2.0/python/src/sdk/models/operations/getresourceinfo.py
import dataclasses
from typing import Optional
from ..shared import resourceresponse as shared_resourceresponse


@dataclasses.dataclass
class GetResourceInfoQueryParams:
    resource: str = dataclasses.field(metadata={'query_param': { 'field_name': 'resource', 'style': 'form', 'explode': True }})
    include: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'include', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetResourceInfoHeaders:
    ev_access_token: str = dataclasses.field(metadata={'header': { 'field_name': 'ev-access-token', 'style': 'simple', 'explode': False }})
    ev_api_key: str = dataclasses.field(metadata={'header': { 'field_name': 'ev-api-key', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetResourceInfoRequest:
    headers: GetResourceInfoHeaders = dataclasses.field()
    query_params: GetResourceInfoQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class GetResourceInfoResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    resource_response: Optional[shared_resourceresponse.ResourceResponse] = dataclasses.field(default=None)
    
